qiling/os/linux/fncc.py

```diff
@@ -205,7 +205,7 @@ def x86_stdcall(ql, param_num, params, func, args, kwargs):
     result, param_num = __x86_cc(ql, param_num, params, func, args, kwargs)
 
     # update stack pointer
-    ql.sp = ql.sp + ((param_num + 1) * 4)
+    ql.reg.sp = ql.reg.sp + ((param_num + 1) * 4)
```

Comment:
This should be ql.reg.arch_sp.

x86 might have issue with ESP(32bit), RSP(64bit) and SP(16bit). That is the reason we name it to ql.reg.arch_sp

Reply:
Thx, update it.